staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource.go

```diff
@@ -17,50 +17,151 @@ limitations under the License.
 package fieldmanager
 
 import (
+	"fmt"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal"
 	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
 )
 
-// FindEntries returns all the managed fields which has a specific path
-func FindEntries(entries []metav1.ManagedFieldsEntry, pathParts ...interface{}) ([]metav1.ManagedFieldsEntry, error) {
-	path, err := fieldpath.MakePath(pathParts...)
+// SubresourceHandler manages the conversion of managed fields between a main
+// resource and subresources
+type SubresourceHandler struct {
+	parentEntries           []metav1.ManagedFieldsEntry
+	parentGroupVersion      schema.GroupVersion
+	subresourceGroupVersion schema.GroupVersion
+
+	mapping         mapping
+	parentPath      fieldpath.Path
+	subresourcePath fieldpath.Path
+}
+
+type mapping struct {
+	subresource []interface{}
+	parent      []interface{}
+}
+
+var mappings map[schema.GroupVersion]mapping = map[schema.GroupVersion]mapping{
+	schema.GroupVersion{Group: "apps", Version: "v1"}: mapping{
+		subresource: []interface{}{"spec", "replicas"},
+		parent:      []interface{}{"spec", "replicas"},
+	},
+}
+
+// NewSubresourceHandler creates a new SubresourceHandler
+func NewSubresourceHandler(parentEntries []metav1.ManagedFieldsEntry, parentGroupVersion, subresourceGroupVersion schema.GroupVersion) (*SubresourceHandler, error) {
+	h := SubresourceHandler{
+		parentEntries:           parentEntries,
+		parentGroupVersion:      parentGroupVersion,
+		subresourceGroupVersion: subresourceGroupVersion,
+	}
+	m, ok := mappings[parentGroupVersion]
```

Comment:
Take this as a parameter probably

Reply:
What are you referring to?

Reply:
Nvm I think I got it, the mappings belong to the storage and passed as parameter.